src/main/java/helper/GoogleUserMapper.java

```diff
@@ -0,0 +1,33 @@
+package helper;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Base64;
+
+public class GoogleUserMapper {
+    public String getUsernameFromEmail(String email) {
+        int usernameEndsAt = email.indexOf('@');
+        if(usernameEndsAt == -1) {
+            return null;
+        }
+        return email.substring(0, usernameEndsAt);
+    }
+
+    public String getPictureFromUrl(String pictureUrl) throws IOException {
+        URL imageUrl = new URL(pictureUrl);
+        URLConnection connection = imageUrl.openConnection();
+        InputStream inputStream = connection.getInputStream();
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
```

Comment:
We are just reading the first 1024 bytes, i.e. 1 KB. I don't think that's enough for a picture. :)

Reply:
We are reading in blocks of 1KB, should I increase this block size? What do you suggest would be a good block size?

Reply:
Oh sorry, I didn't realise this was block by block read. My bad.